### ISSUE TITLE
Need execulable errors to exit scripts

### ISSUE THREAD

**Issue by robbietuk** [UCL/STIR-GATE-Connection#49]
Some scripts can error due to bad inputs (or older versions of STIR). These scripts will continue, even though an output of the error'd function is required later. This leads to much unnessisery computation in some cases and makes debugging harder. Example script I am talking about is: https://github.com/UCL/STIR-GATE-Connection/blob/f3c5f8f49a24ed2dd9782400100102a8cf71f1ef/VoxelisedSimulation/SubScripts/EstimateRandomsFromDelayed.sh#L35-L37 
where `apply_normfactors3D ` fails due to an older version of STIR. Yet the script continues, but there are other examples.

**Comment by KrisThielemans** [comment]
easy thing is to use `set -e` at the start. It does make the script stop whenever there's an error (that isn't inside a pipeline, etc). It does exit abruptly though without any message. We can switch it to `bash` and then do something like this
https://github.com/UCL/STIR/blob/7b4c27470ad175a80f29e9aabe945b6831a4f159/scripts/estimate_scatter.sh#L104-L105
```sh
set -e # exit on error
trap "echo ERROR in $0" ERR
```

Obviously better is to do
```sh
if apply_normfactors3D $OutputFilename $factors ones.hs 1 1 $eff_iters
then
  : #OK
else
  echo "Error over here" >2;exit 1
fi
```
but it's much more verbose. In nearly all Bourne shells implementations you can shorten this to
```sh
if ! apply_normfactors3D $OutputFilename $factors ones.hs 1 1 $eff_iters
then
  echo "Error over here" >2;exit 1
fi
```
but apparently it isn't 100% portable 😢 (although I cannot immediately find a link to that info anymore)

**Comment by robbietuk** [comment]
Is there anything wrong with
```
apply_normfactors3D $OutputFilename $factors ones.hs 1 1 $eff_iters
if [ $? -eq 0 ]; then
    echo "apply_normfactors3D OK"
else
    echo "apply_normfactors3D FAIL"
    exit 1
fi

```

**Comment by KrisThielemans** [comment]
sure, that's fine and even more verbose 😄

**Comment by robbietuk** [comment]
Resolved with 3fef6c4
